drivers/flash/soc_flash_nrf.c

```diff
@@ -420,7 +441,10 @@ static int write_in_timeslice(off_t addr, const void *data, size_t len)
 		.len = len,
 		.enable_time_limit = 1, /* enable time limit */
 		.interval = FLASH_INTERVAL_WRITE,
-		.slot = FLASH_SLOT_WRITE
+		.slot = FLASH_SLOT_WRITE,
+#if defined(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)
+		.flash_addr_next = addr
```

Comment:
It seems unneeded here, as `write_op` does not use this field.
Instead, you probably should initialize this field when preparing `erase()` (see [line 608](https://github.com/zephyrproject-rtos/zephyr/blob/d660b0260274a77c3b6527db754ceed7d3a4fdd1/drivers/flash/soc_flash_nrf.c#L608)), so that `nrfx_nvmc_page_partial_erase_init()` gets called at the beginning of the loop in `erase_op()`.

Reply:
Good point. I actually cp the lines to wrong function so it should have been in erase for the non timesliced version. Thx!